lib/presentation/widget/shared_space_details/add_shared_space_member/add_shared_space_member_viewmodel.dart

```diff
@@ -125,13 +131,46 @@ class AddSharedSpaceMemberViewModel extends BaseViewModel {
     });
   }
 
-  OnlineThunkAction _getSharedSpaceRolesAction() {
+  OnlineThunkAction _updateSharedSpaceMemberRoleAction(
+      SharedSpaceId sharedSpaceId, AccountId accountId, SharedSpaceRoleName newRole) {
     return OnlineThunkAction((Store<AppState> store) async {
-      final roles = (await _getAllSharedSpaceRolesInteractor.execute())
-          .map((result) => result is SharedSpaceRolesViewState ? result.roles : null)
-          .getOrElse(() => []);
+      final rolesList = store.state.sharedSpaceState.rolesList;
+      if (rolesList.isEmpty) {
+        store.dispatch(UpdateSharedSpaceMembersAction(
+            Left<Failure, Success>(UpdateSharedSpaceMemberFailure(RolesListNotFound()))));
+        return;
+      }
+
+      final role = rolesList.firstWhere((_role) => _role.name == newRole, orElse: () => null);
+      if (role == null) {
+        store.dispatch(UpdateSharedSpaceMembersAction(
+            Left<Failure, Success>(UpdateSharedSpaceMemberFailure(SelectedRoleNotFound()))));
+        return;
+      }
 
-      store.dispatch(SharedSpaceGetSharedSpaceRolesListAction(roles));
+      await _updateSharedSpaceMemberInteractor
+          .execute(sharedSpaceId,
+          UpdateSharedSpaceMemberRequest(accountId, sharedSpaceId, role.sharedSpaceRoleId))
+          .then((result) => result.fold(
+              (failure) => store.dispatch(UpdateSharedSpaceMembersAction(
+              Left<Failure, Success>(UpdateSharedSpaceMemberFailure(UpdateRoleFailed())))),
+              (success) {
+            store.dispatch(_updateMembersInSharedSpaceDetails(success is UpdateSharedSpaceMemberViewState ? success.member : null));
+          }));
+    });
+  }
+
+  ThunkAction<AppState> _updateMembersInSharedSpaceDetails(SharedSpaceMember member) {
```

Comment:
I think this function should be removed. Use it right away `store.dispatch(SharedSpaceDetailsUpdateSharedSpaceMembersAction(member));`

Reply:
I did that before. But I want to separate to a new function for other devs can understand easier. This stuff helps to sync the member list even offline